File: pong.js
const canvas = document.getElementById("pongCanvas");
const ctx = canvas.getContext("2d");

// Create the paddles
const paddleWidth = 10, paddleHeight = 100;
let leftPaddleY = canvas.height / 2 - paddleHeight / 2, rightPaddleY = canvas.height / 2 - paddleHeight / 2;

// Ball properties
let ballRadius = 10;
let ballX = canvas.width / 2, ballY = canvas.height / 2;
let ballSpeedX = 4, ballSpeedY = 4;

// Player scores
let playerAScore = 0, playerBScore = 0;

// Control paddles using keys
let leftPaddleUp = false, leftPaddleDown = false, rightPaddleUp = false, rightPaddleDown = false;

document.addEventListener("keydown", (e) => {
    if (e.key === "w") leftPaddleUp = true;
    if (e.key === "s") leftPaddleDown = true;
    if (e.key === "ArrowUp") rightPaddleUp = true;
    if (e.key === "ArrowDown") rightPaddleDown = true;
});

document.addEventListener("keyup", (e) => {
    if (e.key === "w") leftPaddleUp = false;
    if (e.key === "s") leftPaddleDown = false;
    if (e.key === "ArrowUp") rightPaddleUp = false;
    if (e.key === "ArrowDown") rightPaddleDown = false;
});

// Draw the paddles
function drawPaddles() {
    ctx.fillStyle = "white";
    ctx.fillRect(0, leftPaddleY, paddleWidth, paddleHeight);  // Left paddle
    ctx.fillRect(canvas.width - paddleWidth, rightPaddleY, paddleWidth, paddleHeight);  // Right paddle
}

// Draw the ball
function drawBall() {
    ctx.beginPath();
    ctx.arc(ballX, ballY, ballRadius, 0, Math.PI * 2);
    ctx.fillStyle = "red";
    ctx.fill();
    ctx.closePath();
}

// Draw the scores
function drawScores() {
    ctx.font = "24px Arial";
    ctx.fillText(`Player A: ${playerAScore}`, canvas.width / 4, 30);
    ctx.fillText(`Player B: ${playerBScore}`, (canvas.width / 4) * 3, 30);
}

// Move the paddles
function movePaddles() {
    if (leftPaddleUp && leftPaddleY > 0) leftPaddleY -= 5;
    if (leftPaddleDown && leftPaddleY < canvas.height - paddleHeight) leftPaddleY += 5;
    if (rightPaddleUp && rightPaddleY > 0) rightPaddleY -= 5;
    if (rightPaddleDown && rightPaddleY < canvas.height - paddleHeight) rightPaddleY += 5;
}

// Move the ball
function moveBall() {
    ballX += ballSpeedX;
    ballY += ballSpeedY;

    // Ball collisions with top and bottom
    if (ballY - ballRadius < 0 || ballY + ballRadius > canvas.height) {
        ballSpeedY = -ballSpeedY;
    }

    // Ball collisions with paddles
    if (ballX - ballRadius < paddleWidth && ballY > leftPaddleY && ballY < leftPaddleY + paddleHeight) {
        ballSpeedX = -ballSpeedX;
    }

    if (ballX + ballRadius > canvas.width - paddleWidth && ballY > rightPaddleY && ballY < rightPaddleY + paddleHeight) {
        ballSpeedX = -ballSpeedX;
    }

    // Ball out of bounds (scoring)
    if (ballX - ballRadius < 0) {
        playerBScore++;
        resetBall();
    }

    if (ballX + ballRadius > canvas.width) {
        playerAScore++;
        resetBall();
    }
}

// Reset the ball to the center
function resetBall() {
    ballX = canvas.width / 2;
    ballY = canvas.height / 2;
    ballSpeedX = -ballSpeedX;
    ballSpeedY = 4 * (Math.random() < 0.5 ? 1 : -1);
}

// Main game loop
function gameLoop() {
    ctx.clearRect(0, 0, canvas.width, canvas.height);

    drawPaddles();
    drawBall();
    drawScores();

    movePaddles();
    moveBall();

    requestAnimationFrame(gameLoop);
}

// Start the game loop
gameLoop();

File: pong.py
import turtle as t
import os  # To play sound

# Initialize scores
playerAscore = 0
playerBscore = 0

# Create a window and screen setup
window = t.Screen()
window.title("Pong Game with Graphics")
window.bgcolor("black")
window.setup(width=800, height=600)
window.tracer(0)

# Create the top and bottom borders for the Pong game
border = t.Turtle()
border.speed(0)
border.color("white")
border.penup()
border.goto(-390, 290)
border.pendown()
border.pensize(3)
border.goto(390, 290)
border.goto(390, -290)
border.goto(-390, -290)
border.goto(-390, 290)

# Creating the left paddle
leftpaddle = t.Turtle()
leftpaddle.speed(0)
leftpaddle.shape("square")
leftpaddle.color("white")
leftpaddle.shapesize(stretch_wid=5, stretch_len=1)
leftpaddle.penup()
leftpaddle.goto(-350, 0)

# Creating the right paddle
rightpaddle = t.Turtle()
rightpaddle.speed(0)
rightpaddle.shape("square")
rightpaddle.color("white")
rightpaddle.shapesize(stretch_wid=5, stretch_len=1)
rightpaddle.penup()
rightpaddle.goto(350, 0)

# Code for creating the ball
ball = t.Turtle()
ball.speed(0)
ball.shape("circle")
ball.color("red")
ball.penup()
ball.goto(0, 0)
ballxdirection = 0.2  # Horizontal direction of the ball
ballydirection = 0.2  # Vertical direction of the ball

# Code for creating pen for scorecard update
pen = t.Turtle()
pen.speed(0)
pen.color("white")
pen.penup()
pen.hideturtle()
pen.goto(0, 260)
pen.write("Player A: {}  Player B: {}".format(playerAscore, playerBscore), align="center", font=('Arial', 24, 'normal'))

# Code for moving the left paddle
def leftpaddleup():
    y = leftpaddle.ycor()
    if y < 250:  # Prevent moving out of the screen
        y += 20
    leftpaddle.sety(y)

def leftpaddledown():
    y = leftpaddle.ycor()
    if y > -240:  # Prevent moving out of the screen
        y -= 20
    leftpaddle.sety(y)

# Code for moving the right paddle
def rightpaddleup():
    y = rightpaddle.ycor()
    if y < 250:  # Prevent moving out of the screen
        y += 20
    rightpaddle.sety(y)

def rightpaddledown():
    y = rightpaddle.ycor()
    if y > -240:  # Prevent moving out of the screen
        y -= 20
    rightpaddle.sety(y)

# Assign keys to control paddles
window.listen()
window.onkeypress(leftpaddleup, 'w')
window.onkeypress(leftpaddledown, 's')
window.onkeypress(rightpaddleup, 'Up')
window.onkeypress(rightpaddledown, 'Down')

# Game loop
while True:
    window.update()

    # Moving the ball
    ball.setx(ball.xcor() + ballxdirection)
    ball.sety(ball.ycor() + ballydirection)

    # Border collision setup
    if ball.ycor() > 290:
        ball.sety(290)
        ballydirection *= -1
    if ball.ycor() < -290:
        ball.sety(-290)
        ballydirection *= -1

    # Ball hitting the right border (Player A scores)
    if ball.xcor() > 390:
        ball.goto(0, 0)
        ballxdirection *= -1
        playerAscore += 1
        pen.clear()
        pen.write("Player A: {}  Player B: {}".format(playerAscore, playerBscore), align="center", font=('Arial', 24, 'normal'))
        os.system("afplay wallhit.wav&")  # Sound effect when ball crosses right side

    # Ball hitting the left border (Player B scores)
    if ball.xcor() < -390:
        ball.goto(0, 0)
        ballxdirection *= -1
        playerBscore += 1
        pen.clear()
        pen.write("Player A: {}  Player B: {}".format(playerAscore, playerBscore), align="center", font=('Arial', 24, 'normal'))
        os.system("afplay wallhit.wav&")  # Sound effect when ball crosses left side

    # Handling the collisions with paddles
    if (ball.xcor() > 340) and (ball.xcor() < 350) and (ball.ycor() < rightpaddle.ycor() + 40 and ball.ycor() > rightpaddle.ycor() - 40):
        ball.setx(340)
        ballxdirection *= -1
        os.system("afplay paddle.wav&")  # Sound effect when ball hits the right paddle

    if (ball.xcor() < -340) and (ball.xcor() > -350) and (ball.ycor() < leftpaddle.ycor() + 40 and ball.ycor() > leftpaddle.ycor() - 40):
        ball.setx(-340)
        ballxdirection *= -1
        os.system("afplay paddle.wav&")  # Sound effect when ball hits the left paddle
